include/range/v3/at.hpp

```diff
@@ -30,7 +31,20 @@ namespace ranges
         {
             /// \return `begin(rng)[n]`
             template<typename Rng,
-                CONCEPT_REQUIRES_(RandomAccessRange<Rng>())>
+                CONCEPT_REQUIRES_(RandomAccessRange<Rng>() and
+                                  SizedRange<Rng>())>
+            RANGES_CXX14_CONSTEXPR
+            auto operator()(Rng &&rng, range_difference_type_t<Rng> n) const ->
+                decltype(begin(rng)[n])
+            {
+                RANGES_EXPECT(static_cast<std::size_t>(n) < ranges::size(rng));
+                return begin(rng)[n];
+            }
+
+            /// \return `begin(rng)[n]`
+            template<typename Rng,
+                CONCEPT_REQUIRES_(RandomAccessRange<Rng>() and
+                                  not SizedRange<Rng>())>
```

Comment:
`RANGES_EXPECT(static_cast<std::size_t>(n) < ranges::distance(rng));`

`ranges::distance` returns `difference_type_t`, there's no need for the cast.